trunk/source/TreeAutomizer/src/de/uni_freiburg/informatik/ultimate/plugins/generator/treeautomizer/TreeAutomizerChcScript.java

```diff
@@ -0,0 +1,175 @@
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import de.uni_freiburg.informatik.ultimate.automata.AutomataLibraryException;
+import de.uni_freiburg.informatik.ultimate.core.lib.results.TimeoutResult;
+import de.uni_freiburg.informatik.ultimate.core.model.results.IResult;
+import de.uni_freiburg.informatik.ultimate.core.model.services.ILogger;
+import de.uni_freiburg.informatik.ultimate.core.model.services.IUltimateServiceProvider;
+import de.uni_freiburg.informatik.ultimate.lib.chc.Derivation;
+import de.uni_freiburg.informatik.ultimate.lib.chc.HcSymbolTable;
+import de.uni_freiburg.informatik.ultimate.lib.chc.HornAnnot;
+import de.uni_freiburg.informatik.ultimate.lib.chc.HornClause;
+import de.uni_freiburg.informatik.ultimate.lib.chc.IChcScript;
+import de.uni_freiburg.informatik.ultimate.lib.chc.results.ChcSatResult;
+import de.uni_freiburg.informatik.ultimate.lib.chc.results.ChcUnsatResult;
+import de.uni_freiburg.informatik.ultimate.lib.smtlibutils.ManagedScript;
+import de.uni_freiburg.informatik.ultimate.logic.Model;
+import de.uni_freiburg.informatik.ultimate.logic.Script;
+import de.uni_freiburg.informatik.ultimate.logic.Script.LBool;
+import de.uni_freiburg.informatik.ultimate.plugins.generator.traceabstraction.preferences.TAPreferences;
+import de.uni_freiburg.informatik.ultimate.plugins.generator.treeautomizer.graph.TreeAutomizerCEGAR;
+
+public class TreeAutomizerChcScript implements IChcScript {
+	private static final String DUMMY_FILENAME = TreeAutomizerChcScript.class.getSimpleName();
+
+	private final IUltimateServiceProvider mServices;
+	private final ManagedScript mMgdScript;
+	private final TAPreferences mPrefs;
+	private final ILogger mLogger;
+
+	private boolean mProduceUnsatCores = false;
+
+	private LBool mLastResult = null;
+	private Set<HornClause> mUnsatCore;
+
+	public TreeAutomizerChcScript(final IUltimateServiceProvider services, final ManagedScript mgdScript,
+			final TAPreferences prefs) {
+		mServices = services;
+		mMgdScript = mgdScript;
+		mPrefs = prefs;
+
+		mLogger = mServices.getLoggingService().getLogger(TreeAutomizerChcScript.class);
+	}
+
+	@Override
+	public Script getScript() {
+		return mMgdScript.getScript();
+	}
+
+	@Override
+	public LBool solve(final HcSymbolTable symbolTable, final List<HornClause> system) {
+		return solve(mServices, symbolTable, system);
+	}
+
+	@Override
+	public LBool solve(final HcSymbolTable symbolTable, final List<HornClause> system, final long timeout) {
+		final var timer = mServices.getProgressMonitorService().getChildTimer(timeout);
+		final var services = mServices.getProgressMonitorService().registerChildTimer(mServices, timer);
+		return solve(services, symbolTable, system);
+	}
+
+	private LBool solve(final IUltimateServiceProvider services, final HcSymbolTable symbolTable,
+			final List<HornClause> system) {
+		reset();
+
+		// TODO missing parameter: category info
```

Comment:
?

Reply:
The last argument of the `HornAnnot` constructor in the next line.

Reply:
I just dont know if this is an issue and what the parameter does. But if it is ok to leave it like that, just leave it.

Reply:
Resolved in ef5fd2516e